core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -379,6 +419,12 @@ Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
                 }
                 return fields;
             }
+
+            private void validateParsed(Double value) {
```

Comment:
please make it a double rather than Double and use Double.isFinite(value) == false below